modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteDiscoDataHandlingInNewClusterTest.java

```diff
@@ -135,31 +161,50 @@
      * See related ticket <a href="https://issues.apache.org/jira/browse/IGNITE-10878">IGNITE-10878</a>.
      */
     @Test
-    public void testNewClusterFiltersDiscoveryDataReceivedFromStoppedCluster() throws Exception {
+    public void testNewClusterIgnoresDiscoDataFromStopped_SameStaticConfig() throws Exception {
         IgniteEx ig0 = startGrid(NODE_1_CONS_ID);
 
         prepareDynamicCaches(ig0);
 
         IgniteEx ig1 = startGrid(NODE_2_CONS_ID);
 
-        verifyCachesAndGroups(ig1);
+        verifyCachesAndGroups(ig1, Arrays.asList(new String[] {STATIC_CACHE_NAME_1, STATIC_CACHE_NAME_3}));
 
         IgniteEx ig2 = startGrid(NODE_3_CONS_ID);
 
-        verifyCachesAndGroups(ig2);
+        verifyCachesAndGroups(ig2, Arrays.asList(new String[] {STATIC_CACHE_NAME_1, STATIC_CACHE_NAME_3}));
 
         IgniteEx client = startGrid("client01");
 
-        verifyCachesAndGroups(client);
+        verifyCachesAndGroups(client, Arrays.asList(new String[] {STATIC_CACHE_NAME_1, STATIC_CACHE_NAME_3}));
+    }
+
+    /**
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testNewClusterIgnoresDiscoDataFromStopped_DifferentStaticConfig() throws Exception {
+        IgniteEx ig0 = startGrid(NODE_1_CONS_ID);
+
+        prepareDynamicCaches(ig0);
+
+        IgniteEx ig1 = startGrid(NODE_4_DIFF_STATIC_CONFIG);
+
+        verifyCachesAndGroups(ig1, Arrays.asList(new String[] {STATIC_CACHE_NAME_1, STATIC_CACHE_NAME_2}));
+
+        IgniteEx client = startGrid(CLIENT_WITH_DIFF_CONFIG);
+
+        verifyCachesAndGroups(client, Arrays.asList(new String[] {STATIC_CACHE_NAME_1, STATIC_CACHE_NAME_2}));
     }
 
     /** */
-    private void verifyCachesAndGroups(IgniteEx ignite) {
+    private void verifyCachesAndGroups(IgniteEx ignite, Collection<String> cacheNames) {
         Map<String, DynamicCacheDescriptor> caches = ignite.context().cache().cacheDescriptors();
 
-        assertEquals(2, caches.size());
-        caches.keySet().contains(GridCacheUtils.UTILITY_CACHE_NAME);
-        caches.keySet().contains(STATIC_CACHE_NAME);
+        assertEquals(3, caches.size());
```

Comment:
I think it's safer to use cacheNames.size() + 1

Reply:
Fixed.